src/buffer.h

```diff
@@ -42,15 +44,17 @@ struct Buffer {
     size_t rx_bytes;
 };
 
-struct Buffer *new_buffer(size_t, struct ev_loop *);
+struct Buffer *new_buffer(int, size_t, struct ev_loop *);
 void free_buffer(struct Buffer *);
 
 ssize_t buffer_recv(struct Buffer *, int, int, struct ev_loop *);
+ssize_t buffer_recvmsg(struct Buffer *, int, struct msghdr *, int, struct ev_loop *);
 ssize_t buffer_send(struct Buffer *, int, int, struct ev_loop *);
+ssize_t buffer_sendmsg(struct Buffer *, int, struct msghdr *, int, struct ev_loop *);
 ssize_t buffer_read(struct Buffer *, int);
 ssize_t buffer_write(struct Buffer *, int);
 ssize_t buffer_resize(struct Buffer *, size_t);
-size_t buffer_peek(const struct Buffer *, void *, size_t);
+size_t buffer_peek(struct Buffer *, void *, size_t);
```

Comment:
I thought `buffer_peek()` took a `const struct Buffer *`, and was trying to figure out how we could use mutate it before `setup_read_iov()`. Maybe we teach `setup_read_iov()` and `setup_write_iov()` the SOCK_DGRAM length prefix. Will that make the higher functions easier?